nimare/meta/cbma/base.py

```diff
@@ -150,8 +150,18 @@ def _collect_ma_maps(self, coords_key="coordinates", maps_key="ma_maps", fname_i
                     mode="w+",
                     shape=unmasked_shape,
                 )
-                for i, f in enumerate(self.inputs_[maps_key]):
-                    ma_maps[i, :] = self.masker.transform(f)
+
+                # perform transform on chunks of the input maps
+                chunk_size = determine_chunk_size(limit, temp)
+                map_chunks = [
+                    self.inputs_[maps_key][i : i + chunk_size]
+                    for i in range(0, len(self.inputs_[maps_key]), chunk_size)
+                ]
+                idx = 0
+                for map_chunk in map_chunks:
+                    end_idx = idx + len(map_chunk)
+                    ma_maps[idx:end_idx, :] = self.masker.transform(map_chunk)
+                    idx = end_idx
```

Comment:
I'm having this issue with the `CorrelationDistributionDecoder`, and I think it could apply to any case where we're masking many images at once. WDYT of moving this into a new function?

Example signature:

```python
masked_data = safe_transform(imgs, masker, memory_limit="1gb", dtype="auto")
```

`memory_limit` would feed into `determine_chunk_size`, of course, and we could have `dtype` to let the function coerce the data to a given datatype.